feat: add `math/base/special/fast/hypotf`

nakul-krishnakumar opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: missing_dependencies
  - task: lint_c_examples status: missing_dependencies
  - task: lint_c_benchmarks status: missing_dependencies
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds implementation for `math/base/special/fast/hypotf` .

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

nakul-krishnakumar committed:
feat: add `math/base/special/fast/hypotf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/fast/hypotf/README.md:
````diff
@@ -0,0 +1,218 @@
+-   For a sufficiently large `x` and/or `y`, computing the hypotenuse will overflow.
+
+    ```javascript
+    var h = hypotf( 1.0e154, 1.0e154 );
````
These values are outside the range of double-precision. You need to decrease the values here and increase the values at L72, as those are also beyond single-precision range.

kgryte commented on lib/node_modules/@stdlib/math/base/special/fast/hypotf/docs/repl.txt:
```diff
@@ -0,0 +1,33 @@
+    13.0
+
+    // For a sufficiently large `x` and/or `y`, the function overflows:
+    > h = {{alias}}( 1.0e154, 1.0e154 )
```
Same thing here and below. The values need to be adjusted to within single-precision range.

kgryte commented on lib/node_modules/@stdlib/math/base/special/fast/hypotf/test/test.js:
```diff
@@ -0,0 +1,94 @@
+});
+
+tape( 'the function can overflow', function test( t ) {
+	var h = hypotf( 1.0e308, 1.0e308 );
```
These values and L91 need to be adjusted to within single-precision range.

kgryte commented on lib/node_modules/@stdlib/math/base/special/fast/hypotf/test/test.native.js:
```diff
@@ -0,0 +1,103 @@
+});
+
+tape( 'the function can overflow', opts, function test( t ) {
+	var h = hypotf( 1.0e308, 1.0e308 );
```
These values and L100 need to be adjusted to within single-precision range.

nakul-krishnakumar replied:
Understood, thank you for the review!

kgryte reviewed (changes requested):
Other than some incorrect values, this PR is looking good.

nakul-krishnakumar committed:
chore: change according to code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM

anandkaranubc commented:
Hi @nakul-krishnakumar, thanks for your PR! A few changes were missing, which have been addressed here: [https://github.com/stdlib-js/stdlib/compare/18b237247617...fe96d973306f](https://github.com/stdlib-js/stdlib/compare/18b237247617...fe96d973306f). Just wanted to put this on your radar.

The main update was ensuring that input values and arithmetic operations are properly cast to `Float32`. If you’d like to learn more about how we handle single-precision in JS and why it’s important for single-precision implementations, this [section](https://github.com/stdlib-js/stdlib/wiki/GSoC-2025-%E2%80%90-Karan-Anand#single-precision-js-support) might be a helpful read.

Cheers!

cc: @kgryte

nakul-krishnakumar commented:
> Hi @nakul-krishnakumar, thanks for your PR! A few changes were missing, which have been addressed here: [18b2372...fe96d97](https://github.com/stdlib-js/stdlib/compare/18b237247617...fe96d973306f). Just wanted to put this on your radar.
> 
> The main update was ensuring that input values and arithmetic operations are properly cast to `Float32`. If you’d like to learn more about how we handle single-precision in JS and why it’s important for single-precision implementations, this [section](https://github.com/stdlib-js/stdlib/wiki/GSoC-2025-%E2%80%90-Karan-Anand#single-precision-js-support) might be a helpful read.
> 
> Cheers!
> 
> cc: @kgryte

Thank you for the review! I will make sure I do the necessary changes while working with single-precision implementations in the future. :)
